src/tree/namespace/tests/test_hasher.cairo

```diff
@@ -0,0 +1,121 @@
+use core::option::OptionTrait;
+use core::traits::Into;
+use alexandria_bytes::BytesTrait;
+use blobstream_sn::tree::namespace::hasher::{leaf_digest, node_digest};
+use blobstream_sn::tree::namespace::merkle_tree::{Namespace, NamespaceNode, namespace_node_eq};
+use blobstream_sn::tree::consts::{parity_share_namespace};
+use debug::PrintTrait;
+
+#[test]
+fn leaf_digest_empty_test() {
+    // nid.print();
+
+    let bytesval: u128 = 0x00000000000000000000000000000000000000000000000000000000;
+
+    let nid: Namespace = Namespace { version: 0x00, id: bytesval.into() };
+
+    let expected: NamespaceNode = NamespaceNode {
+        min: nid,
+        max: nid,
+        digest: 0x0679246d6c4216de0daa08e5523fb2674db2b6599c3b72ff946b488a15290b62
+    };
+
+    let data = BytesTrait::zero(0);
+
+    let node: NamespaceNode = leaf_digest(nid, @data);
+
+    let res = namespace_node_eq(node, expected);
+    assert(res, 'Not equal to expected digest');
```

Comment:
The latest pattern for doing asserts in tests is to use the `assert!` macro, which gives better printing outputs with strings.